<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>XVideoUIClass</class>
 <widget class="QWidget" name="XVideoUIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>833</width>
    <height>797</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>XVideoUI</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#XVideoUIClass
{
	
	background-color: rgb(53, 53, 53);

}
QPushButton:hover
 { 
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(120, 120, 120, 255), stop:0.495 rgba(39, 39, 39, 255), stop:0.505 rgba(39,39, 39, 255), stop:1 rgba(120, 120,120, 255));
 border: none;
border-radius:5px;
color: rgb(255, 255, 255);
font: 75 12pt &quot;黑体&quot;;
 }
QPushButton:!hover
 { 
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(39, 39, 39, 255), stop:0.495 rgba(120, 120, 120, 255), stop:0.505 rgba(120,120, 120, 255), stop:1 rgba(39, 39,39, 255));
 border: none;
border-radius:5px;
color: rgb(255, 255, 255);
font: 75 12pt &quot;黑体&quot;;
 }

QLabel
{
	
	color: rgb(255, 255, 255);
	
	font: 75 12pt &quot;黑体&quot;;
}

#playButton{background-color: rgba(255, 255, 255, 0);}
#playButton:!hover{image: url(:/XVideoUI/Resources/PlayN.png);}
#playButton:hover{image: url(:/XVideoUI/Resources/PlayH.png);}

#playButton:pressed{image: url(:/XVideoUI/Resources/PlayP.png);}

#pauseButton{background-color: rgba(255, 255, 255,0);}
#pauseButton:!hover{image: url(:/XVideoUI/Resources/PauseN.png);}

#pauseButton:hover{image: url(:/XVideoUI/Resources/PauseH.png);}
#pauseButton:pressed{image: url(:/XVideoUI/Resources/PauseP.png);}

</string>
  </property>
  <widget class="QPushButton" name="closeButton">
   <property name="geometry">
    <rect>
     <x>760</x>
     <y>0</y>
     <width>41</width>
     <height>31</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-image: url(:/XVideoUI/close.png);</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="flat">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="minButton">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>0</y>
     <width>41</width>
     <height>31</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-image: url(:/XVideoUI/mini.png);</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="flat">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="XVideoWidget" name="src_video">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>40</y>
     <width>380</width>
     <height>280</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="OpenButton">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>340</y>
     <width>71</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
  </widget>
  <widget class="QSlider" name="playSlider">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>350</y>
     <width>381</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="XVideoWidget" name="des_video">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>40</y>
     <width>380</width>
     <height>280</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="bright_label">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>390</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>12</pointsize>
     <weight>9</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>亮度[0-100]</string>
   </property>
  </widget>
  <widget class="QLabel" name="contrast_label">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>420</y>
     <width>141</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>对比度[1.0-3.0]</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="bright">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>390</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="setButton">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>390</y>
     <width>71</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>设置</string>
   </property>
  </widget>
  <widget class="QPushButton" name="exportButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>340</y>
     <width>101</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>导出</string>
   </property>
  </widget>
  <widget class="QPushButton" name="playButton">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>400</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="pauseButton">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>400</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="rotate_label">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>450</y>
     <width>101</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>12</pointsize>
     <weight>9</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>图像旋转</string>
   </property>
  </widget>
  <widget class="QComboBox" name="rotate">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>460</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>0</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>90</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>180</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>270</string>
    </property>
   </item>
  </widget>
  <widget class="QDoubleSpinBox" name="contrast">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>420</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>1.000000000000000</double>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
   <property name="value">
    <double>1.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="fliplabel">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>480</y>
     <width>101</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>12</pointsize>
     <weight>9</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>图像镜像</string>
   </property>
  </widget>
  <widget class="QComboBox" name="flip">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>490</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>不处理</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>左右镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下左右镜像</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="whlabel">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>510</y>
     <width>101</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>12</pointsize>
     <weight>9</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>图像尺寸W,H</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="width_spinBox">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>520</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="height_spinBox">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>520</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QLabel" name="py">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>550</y>
     <width>91</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>12</pointsize>
     <weight>9</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>图像金字塔（高斯、拉普拉斯）</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="pydown">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>550</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="pyup">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>550</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QLabel" name="Clip">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>580</y>
     <width>111</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>12</pointsize>
     <weight>9</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>图像裁剪（x,y,width,height）</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>580</y>
     <width>291</width>
     <height>24</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSpinBox" name="cx">
      <property name="maximum">
       <number>9999</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSpinBox" name="cy">
      <property name="maximum">
       <number>9999</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSpinBox" name="cw">
      <property name="maximum">
       <number>9999</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSpinBox" name="ch">
      <property name="maximum">
       <number>9999</number>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QComboBox" name="color">
   <property name="geometry">
    <rect>
     <x>680</x>
     <y>440</y>
     <width>87</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>RGB图像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>灰度图</string>
    </property>
   </item>
  </widget>
  <widget class="QPushButton" name="MarkButton">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>440</y>
     <width>101</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>水印x,y,a</string>
   </property>
  </widget>
  <widget class="XVideoWidget" name="src2">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>500</y>
     <width>380</width>
     <height>280</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="Blend_Button">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>610</y>
     <width>81</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>融合</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ba">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>630</y>
     <width>70</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="value">
    <double>0.500000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="Merge_Button">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>660</y>
     <width>81</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>合并</string>
   </property>
  </widget>
  <widget class="QSlider" name="left">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>320</y>
     <width>381</width>
     <height>22</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QSlider::handle:horizontal {
background:  rgb(0, 170, 255);
border-radius: 5px;
}
</string>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QSlider" name="right">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>380</y>
     <width>381</width>
     <height>22</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QSlider::handle:horizontal {
background:  rgb(0, 170, 255);
border-radius: 5px;
}
</string>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="value">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>450</y>
     <width>191</width>
     <height>24</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QSpinBox" name="mx"/>
    </item>
    <item>
     <widget class="QSpinBox" name="my"/>
    </item>
    <item>
     <widget class="QDoubleSpinBox" name="ma">
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>XVideoWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>xvideowidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="xvideoui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>closeButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>772</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>789</x>
     <y>666</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>showMinimized()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>735</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>726</x>
     <y>677</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderPressed()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SliderPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>306</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SliderRelease()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>270</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>280</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SetPos(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>135</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Set()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>718</x>
     <y>420</y>
    </hint>
    <hint type="destinationlabel">
     <x>766</x>
     <y>751</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exportButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Export()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>576</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Play()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>420</y>
    </hint>
    <hint type="destinationlabel">
     <x>241</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Pause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>424</y>
    </hint>
    <hint type="destinationlabel">
     <x>141</x>
     <y>751</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MarkButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Mark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>751</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Blend_Button</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Blend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>640</y>
    </hint>
    <hint type="destinationlabel">
     <x>476</x>
     <y>796</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OpenButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>462</x>
     <y>358</y>
    </hint>
    <hint type="destinationlabel">
     <x>486</x>
     <y>677</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Merge_Button</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Merge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>446</x>
     <y>691</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>796</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>left</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Left(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>327</y>
    </hint>
    <hint type="destinationlabel">
     <x>121</x>
     <y>-93</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>right</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Right(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>30</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>40</x>
     <y>927</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Open()</slot>
  <slot>SliderPress()</slot>
  <slot>SliderRelease()</slot>
  <slot>SetPos(int)</slot>
  <slot>Set()</slot>
  <slot>Export()</slot>
  <slot>Play()</slot>
  <slot>Pause()</slot>
  <slot>Mark()</slot>
  <slot>Blend()</slot>
  <slot>Merge()</slot>
  <slot>Left(int)</slot>
  <slot>Right(int)</slot>
 </slots>
</ui>
